### ISSUE TITLE
`Base.runtests(String[])` runs all tests

### ISSUE THREAD

**Issue by sostock** [JuliaLang/julia#44979]
`Base.runtests(tests)` should only[^1] run the specified tests. However, passing an empty vector runs all tests instead of no tests.

[^1]: To be fair, the docstring says “Run the Julia unit tests listed in `tests`”, not “Run *only* the Julia unit tests listed in `tests`”.

**Comment by DilumAluthge** [comment]
I think this is the intended behavior. We should update the docstring to clarify that `Base.runtests([])` is equivalent to `Base.runtests("all")`.

**Comment by StefanKarpinski** [comment]
It's not great from an API perspective. It makes sense on the command line where no arguments is often interpreted as "do the default" but for a programmatic API it's a bad corner case IMO.

**Comment by DilumAluthge** [comment]
> It's not great from an API perspective. It makes sense on the command line where no arguments is often interpreted as "do the default" but for a programmatic API it's a bad corner case IMO.

Yeah, and one reason it works that way currently is that `Base.runtests` basically just forwards to `test/runtests.jl`.

**Comment by DilumAluthge** [comment]
I guess we could make it so that `Base.runtests(x(` throws an error if `isempty(x(`

We'd just have to make sure to not modify the behavior of `make test`, `julia test/runtests.jl`, etc.

**Comment by DilumAluthge** [comment]
What we don't want is for `Base.runtests([])` to run no tests and just exit with exit code zero. There may be people that are currently using `Base.runtests([])` to run all of the tests; I don't want to silently change the meaning of that. I think our options are:

1. Keep the current behavior.
2. Make `Base.runtests([])` throw an error.

**Comment by StefanKarpinski** [comment]
Eh, just leave it then. Seems like too much of a hassle to change it.
